crypto_track/blog.md

```diff
@@ -0,0 +1,574 @@
+        setNotificationLoading(true);
+        const { data, error } = await supabase.from('notifications').select();
+        handleError(error)
+        setNotifications([...data]);
+        setNotificationLoading(false);
+    };
+
+
+```
+
+We also handle displaying the latest cryptocurrency market data by calling the [CoinGecko API](https://www.coingecko.com/en/api/documentation). 
+
+
+```jsx
+
+const onShowCryptoData = async (id) => {
+        setDataLoading(true);
+        try {
+            const res = await fetch(`https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&ids=${id}&price_change_percentage=1h`);
+            const list = await res.json();
+            setMarketData(list[0]);
+            onOpen();
+        }
+        catch (e) {
+            alert(JSON.stringify(e));
+        }
+
+        setDataLoading(false);
+    }
+
+```
+
+> You can see the complete source-code from the `Home` component on [GitHub](https://github.com/eyuelberga/CryptoTrack/blob/main/web/src/Home.jsx).
+
+
+### Part 4: Setting Up a Scheduled Webhook
+
+#### Creating a Webhook
+
+We will now need to create a webhook that will fetch the latest market data and notify subscribers. But we also want it to run periodically so that users don’t miss anything. To do that, we will extend the Postgres database hosted on Supabase to act as a [CRON server](https://supabase.com/blog/postgres-as-a-cron-server).
+
+Let’s get started by creating our webhook. First, we need to create a new Nodejs project. Create a new folder called `cron`, and run the following command:
+
+```sh
+$ yarn init -y
+
+```
+Next, we add our dependencies. We want the server to run on Express(https://expressjs.com/) so we will add that and also the [Courier Javascript client](https://www.npmjs.com/package/@trycourier/courier), the Supabase Javascript client and [axios](https://www.npmjs.com/package/axios).
+
+```
+$ yarn add  @supabase/supabase-js @trycourier/courier axios
+
+```
+
+We need to get our authorization token to set up the Courier client. You can get this token from the Notification details page.
+
+![Courier Notification Detail](./images/courier-notification-detail.png)
+
+
+After retrieving our token we can create an instance of our Courier client:
+
+```js
+
+import { CourierClient } from "@trycourier/courier";
+const courier = CourierClient({ authorizationToken: process.env.COURIER_KEY });
+export default courier;
+```
+
+
+There are a lot of utility functions we need to create for working with the Supabase database and also fetching latest market data. You can see all of them on [GitHub](https://github.com/eyuelberga/CryptoTrack/tree/main/cron/src). The most important one we need to focus on is the `notification.js` file. This holds the logic for how we notify new market changes to subscribers and send emails using the Courier API. 
+
+
+The main function here is `notify`:
+
+```js
+
+const notify = async (supabase) => {
+    // get all notifications
```

Comment:
Love how you commented your code.

Reply:
Thanks. Without them, I would be lost reading my own code the next day.